code/__HELPERS/icons.dm

```diff
@@ -1220,3 +1220,46 @@ GLOBAL_DATUM_INIT(dummySave, /savefile, new("tmp/dummySave.sav")) //Cache of ico
 
 	var/icon/I = getFlatIcon(thing)
 	return icon2html(I, target)
+
+GLOBAL_LIST_EMPTY(transformation_animation_objects)
+
+/atom/movable/proc/transformation_animation(result_appearance,time = 3 SECONDS,transform_overlay=FALSE,reset_after=TRUE)
```

Comment:
add some autodoc and explanation on how to use this else it'll end up going into obscurity